core/block_flyout_inflater.ts

```diff
@@ -0,0 +1,200 @@
+      // Record blocks that were initially disabled.
+      // Do not enable these blocks as a result of capacity filtering.
+      this.permanentlyDisabledBlocks.add(block);
+    } else {
+      this.updateStateBasedOnCapacity(block);
+    }
+
+    // Mark blocks as being inside a flyout.  This is used to detect and
+    // prevent the closure of the flyout if the user right-clicks on such
+    // a block.
+    block.getDescendants(false).forEach((b) => (b.isInFlyout = true));
+    this.addBlockListeners(block);
+
+    return block;
+  }
+
+  createBlock(blockDefinition: BlockInfo, workspace: WorkspaceSvg): BlockSvg {
+    let block;
+    if (blockDefinition['blockxml']) {
+      const xml = (
+        typeof blockDefinition['blockxml'] === 'string'
+          ? utilsXml.textToDom(blockDefinition['blockxml'])
+          : blockDefinition['blockxml']
+      ) as Element;
+      block = Xml.domToBlockInternal(xml, workspace);
+    } else {
+      if (blockDefinition['enabled'] === undefined) {
+        blockDefinition['enabled'] =
+          blockDefinition['disabled'] !== 'true' &&
+          blockDefinition['disabled'] !== true;
+      }
+      if (
+        blockDefinition['disabledReasons'] === undefined &&
+        blockDefinition['enabled'] === false
+      ) {
+        blockDefinition['disabledReasons'] = [MANUALLY_DISABLED];
+      }
+      block = blocks.appendInternal(blockDefinition as blocks.State, workspace);
+    }
+
+    return block as BlockSvg;
+  }
+
+  gapForElement(state: Object, defaultGap: number): number {
+    const blockState = state as BlockInfo;
+    let gap;
+    if (blockState['gap']) {
+      gap = parseInt(String(blockState['gap']));
+    } else if (blockState['blockxml']) {
+      const xml = (
+        typeof blockState['blockxml'] === 'string'
+          ? utilsXml.textToDom(blockState['blockxml'])
+          : blockState['blockxml']
+      ) as Element;
+      gap = parseInt(xml.getAttribute('gap')!);
+    }
+
+    return !gap || isNaN(gap) ? defaultGap : gap;
+  }
+
+  disposeElement(element: IBoundedElement): void {
+    if (!(element instanceof BlockSvg)) return;
+    this.removeListeners(element.id);
+    element.dispose(false, false);
+  }
+
+  protected removeListeners(blockId: string) {
+    const blockListeners = this.listeners.get(blockId) ?? [];
+    blockListeners.forEach((l) => browserEvents.unbind(l));
+    this.listeners.delete(blockId);
+  }
+
+  protected setFlyoutWorkspace(workspace: WorkspaceSvg) {
+    if (this.flyoutWorkspace === workspace) return;
+
+    if (this.flyoutWorkspace) {
+      this.flyoutWorkspace.targetWorkspace?.removeChangeListener(
+        this.capacityFilter,
+      );
+    }
+    this.flyoutWorkspace = workspace;
+    this.flyoutWorkspace.targetWorkspace?.addChangeListener(
+      this.capacityFilter,
+    );
+  }
+
+  private updateStateBasedOnCapacity(block: BlockSvg) {
+    const enable = this.flyoutWorkspace?.targetWorkspace?.isCapacityAvailable(
```

Comment:
Please check [these cases](https://google.github.io/blockly-samples/examples/max-blocks-demo/index.html) to make sure the capacity filtering is still happening the same way, if we don't have automated tests for it.

Reply:
Linked it in and it appears to be.